hs_core/models.py

```diff
@@ -1306,6 +1447,13 @@ def delete(self, using=None):
                 fl.resource_file.delete()
             elif fl.fed_resource_file:
                 fl.fed_resource_file.delete()
+
+            if fl.logical_file is not None:
+                fl.logical_file.delete_metadata()
+                # delete of metadata deletes the logical file (one-to-one relation)
+                # so no need for fl.logical_file.delete()
+                fl.logical_file.metadata.delete()
```

Comment:
```fl.logical_file.delete_metadata() ``` deletes all metadata objects of base type ```AbstractMetaDataElement```

```fl.logical_file.metadata.delete() ``` deletes the metadata object of base type ```AbstractFileMetaData```

I guess I could override the ```metadata.delete()``` so that I don't have to do ```fl.logical_file.delete_metadata() ```